### PR TITLE
IPAM documentation

### PR DESCRIPTION
**Pull request by davidefalcone1**
This PR adds a section on the Liqo Network Manager documentation page that describes the IPAM module and its important role in different situations: remapping of networks, translation of IP addresses of offloaded Pods and mapping of endpoint IP addresses during reflection.

### CODE REVIEW

File: docs/pages/concepts/networking/components/network-manager.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ title: Liqo-Network-Manager
...
+The Liqo Network Manager component manages the creation of the`networkconfigs.net.liqo.io` custom resources for the remote clusters. It also processes the`networkconfigs.net.liqo.io` received from the remote clusters remapping the remote networks if they overlap with any of the address spaces used in the local cluster (See [IPAM](#ipam) for more information). And when the`networkconfigs.net.liqo.io` exchanged by the two clusters are ready the Liqo Network Manager creates a new resource of type `tunnelendpoints.net.liqo.io` which models the network interconnection between the two clusters. The `tunnelendpoints.net.liqo.io` is reconciled by the [Tunnel Operator](../gateway#tunnel-operator) and the [Route Operator](../route#route-operator).
 
 The following diagram illustrates how Liqo Network Manager creates the `tunnelendpoints.net.liqo.io`.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:11]: Add a description of the above diagram. The content is not obvious for a reader not familiar with the matter.

File: docs/pages/concepts/networking/components/network-manager.md [hunk 2]
```diff
@@ -19,3 +19,59 @@ The network parameters between two clusters are exchanged using the`networkconfi
 2. `NCFG 1->2` is replicated to the remote cluster by the `CRDReplicator`, running in Cluster1 who knows how to interact with the API server of Cluster2. Same step done in Cluster2;
 3. In Cluster2 the custom resource `NCFG (1->2)` is processed and its status updated with the NAT information if any. The same is done with `NCFG (2->1)` in Cluster1.
 4. The CRDReplicator running in Cluster1 reflects this changes in the local resource `NCFG(1->2)`. Same step done in Cluster2; The TunnelEndpointCreator in Cluster1 combining the status of `NCFG(1->2)` and the spec of `NCFG (2->1)` it has all the necessary information to create the custom resource `TEP (1-2)` of type `tunnelendpoints.net.liqo.io`.
+
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:27]: 1) There are **5** steps in the diagram, **4** in the text. Please arrange the text accordingly to the 5 steps.
2) Modify the picture so that Liqo components (e.g., CRD replicators) are **within** the proper cluster. All the components should be **inside** their cluster, not outside.

File: docs/pages/concepts/networking/components/network-manager.md [hunk 3]
```diff
@@ -19,3 +19,59 @@ The network parameters between two clusters are exchanged using the`networkconfi
 2. `NCFG 1->2` is replicated to the remote cluster by the `CRDReplicator`, running in Cluster1 who knows how to interact with the API server of Cluster2. Same step done in Cluster2;
 3. In Cluster2 the custom resource `NCFG (1->2)` is processed and its status updated with the NAT information if any. The same is done with `NCFG (2->1)` in Cluster1.
 4. The CRDReplicator running in Cluster1 reflects this changes in the local resource `NCFG(1->2)`. Same step done in Cluster2; The TunnelEndpointCreator in Cluster1 combining the status of `NCFG(1->2)` and the spec of `NCFG (2->1)` it has all the necessary information to create the custom resource `TEP (1-2)` of type `tunnelendpoints.net.liqo.io`.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md]: The reader here has no idea what NCFG contains (the same for TEP). Please make an example, so that the reader understand the flow better.
Important: if not needed, do not report here the entire structure of a NCFG/TEP, but report here only the most significant information in order to give relevant info to the reader, without distracting it with non-necessary details.

File: docs/pages/concepts/networking/components/network-manager.md [hunk 4]
```diff
@@ -1,21 +1,98 @@
...
+1. The network configuration of Cluster1 is saved in a local custom resource (`NCFG(1->2)`) containing the ClusterID of the remote cluster to whom is destined. Same step done in Cluster2;
+2. The `CRDReplicator` running in Cluster1 (which knows how to interact with the API server of Cluster2) replicates `NCFG(1->2)` in the remote cluster. The same step is done in Cluster2;
+3. The custom resource `NCFG(1->2)` in Cluster2 is processed and its status updated with the NAT information (if any). The same is done with `NCFG(2->1)` in Cluster1.
+4. The `CRDReplicator` running in Cluster1 reflects this changes in the local resource `NCFG(1->2)`. Same step done in Cluster2; The TunnelEndpointCreator in Cluster1 combining the status of `NCFG(1->2)` and the spec of `NCFG(2->1)` it has all the necessary information to create the custom resource `TEP(1-2)` of type `tunnelendpoints.net.liqo.io`.
+
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:26]: Please add a small example about the content of NCFG/TEP. The reader has no idea what's contained there, so the flow is hard to follow.
Please list only the relevant information for the reader to understand; if not needed, do not list the complete resource, which may be full of un-necessary details which may distract the reader.

File: docs/pages/concepts/networking/components/network-manager.md [hunk 5]
```diff
@@ -1,21 +1,98 @@
...
+
+### IPAM
+Embedded within the Liqo Network Manager, the IPAM (IP Address Management) is the Liqo module in charge of:
+1. Manage networks and IP addresses, as the name suggests.
+2. Translate IP addresses of offloaded Pods (assigned by foreign clusters) in addresses valid on the home cluster.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:31]:
```suggestion
2. Provide an API that returns to the IP address of an offloaded remote pod (assigned by foreign clusters) in the corresponding IP address that are visible from the home cluster.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 6]
```diff
@@ -1,21 +1,98 @@
...
+1. The network configuration of Cluster1 is saved in a local custom resource (`NCFG(1->2)`) containing the ClusterID of the remote cluster to whom is destined. Same step done in Cluster2;
+2. The `CRDReplicator` running in Cluster1 (which knows how to interact with the API server of Cluster2) replicates `NCFG(1->2)` in the remote cluster. The same step is done in Cluster2;
+3. The custom resource `NCFG(1->2)` in Cluster2 is processed and its status updated with the NAT information (if any). The same is done with `NCFG(2->1)` in Cluster1.
+4. The `CRDReplicator` running in Cluster1 reflects this changes in the local resource `NCFG(1->2)`. Same step done in Cluster2; The TunnelEndpointCreator in Cluster1 combining the status of `NCFG(1->2)` and the spec of `NCFG(2->1)` it has all the necessary information to create the custom resource `TEP(1-2)` of type `tunnelendpoints.net.liqo.io`.
+
+
+### IPAM
+Embedded within the Liqo Network Manager, the IPAM (IP Address Management) is the Liqo module in charge of:
+1. Manage networks and IP addresses, as the name suggests.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:30]: "Managing IP address" is obvious. Be more specific about what this component does.

File: docs/pages/concepts/networking/components/network-manager.md [hunk 7]
```diff
@@ -1,21 +1,98 @@
...
+2. Translate IP addresses of offloaded Pods (assigned by foreign clusters) in addresses valid on the home cluster.
+3. Provide IP addresses for endpoints to the Virtual Kubelet during the [Reflection](../../../offloading/features/api-reflection#Overview).
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:32]: I did not understand, probably because you assume the reader is already familiar with VK and reflection.

File: docs/pages/concepts/networking/components/network-manager.md [hunk 8]
```diff
@@ -1,21 +1,98 @@
...
+
+
+#### Networks and addresses management
+Liqo enables Pods in different (peered) clusters to talk one to another.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:36]:
```suggestion
Liqo enables Pods in different (peered) clusters to talk to each other.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 9]
```diff
@@ -1,21 +1,98 @@
...
+Communicating over the Internet requires involved entities to have an IP address; this is not a problem for Pods, as they are assigned an address by the CNI installed on their cluster.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:37]:
```suggestion
The cluster CNI provider assigns to each Pod a unique IP address, which is valid _within the cluster_ and that allows each Pod to talk to each other, directly, even without the use of a _service_.
Given that Liqo extends a first cluster on the resources shared by a (second) remote cluster, Liqo has to enable Pods in different (peered) clusters to talk to each other directly, even if IP addresses are valid only within the context of the cluster and therefore are not reachable by Pods in peered clusters.
However, pods belonging to _two_ clusters cannot talk directly
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 10]
```diff
@@ -1,21 +1,98 @@
...
+However, these IPs are valid only within the context of the cluster and therefore are not reacheable by Pods in peered clusters.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:38]:
```suggestion
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 11]
```diff
@@ -1,21 +1,98 @@
...
+The [Tunnel Operator](../gateway#tunnel-operator) solves this problem, enabling Pods to communicate not only with those running on their same cluster, but also with those running on peered clusters.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:39]: According to the text, the tunnel operator solves the problem. So, there's no reason why IPAM is involved. Please reformulate in order to make things more clear.

File: docs/pages/concepts/networking/components/network-manager.md [hunk 12]
```diff
@@ -1,21 +1,98 @@
...
+
+
+#### Networks and addresses management
+Liqo enables Pods in different (peered) clusters to talk one to another.
+Communicating over the Internet requires involved entities to have an IP address; this is not a problem for Pods, as they are assigned an address by the CNI installed on their cluster.
+However, these IPs are valid only within the context of the cluster and therefore are not reacheable by Pods in peered clusters.
+The [Tunnel Operator](../gateway#tunnel-operator) solves this problem, enabling Pods to communicate not only with those running on their same cluster, but also with those running on peered clusters. 
+
+But how does the IPAM come into the rescue?
+Liqo enables traffic toward clusters' PodCIDR but does not make any assumption on those address spaces.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:42]:
```suggestion
Liqo enables inter-cluster Pod-to-Pod without making any assumption on the address spaces (i.e., PodCIDR) configured in each cluster.
Therefore, Liqo must work even when the peered clusters have the same PodCIDR, hence the two Pods could have conflicting IP addresses.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 13]
```diff
@@ -1,21 +1,98 @@
...
+It is legitimate to imagine a situation in which two peered clusters have the same PodCIDR: in this case two Pods could be assigned the same IP addresses, creating a conflict.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:43]:
```suggestion
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 14]
```diff
@@ -1,21 +1,98 @@
...
+
+{{% notice info %}}
+The PodCIDR is the network used by the CNI for assigning addresses to Pods.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:46]:
```suggestion
The PodCIDR is the IP addressing space used by the CNI to assign addresses to Pods (e.g., _10.0.0.0/16_).
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 15]
```diff
@@ -1,21 +1,98 @@
...
+The ExternalCIDR is the network used by Liqo for assigning addresses to external resources during Reflection(See more [Here](#reflection-of-external-endpoints)).
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:47]:
```suggestion
The ExternalCIDR is the IP addressing space used by Liqo for assigning addresses to external resources during Reflection (See more [Here](#reflection-of-external-endpoints)).
```

Comment (frisso) [reply docs/pages/concepts/networking/components/network-manager.md:47]: 1) The ExternalCIDR comes here out of the blue, it has never been presented.
2) Not clear what this IP addressing space is. Please improve the definition.

**Review by frisso — CHANGES_REQUESTED** [review]

File: docs/pages/concepts/networking/components/network-manager.md [hunk 16]
```diff
@@ -1,21 +1,98 @@
...
+{{% /notice %}}
+
+The Liqo solution is to check if networks (PodCIDR and ExternalCIDR) of a foreign cluster are already in use or are reserved in the home cluster (these networks can be specified in the `clusterconfigs.config.liqo.io` resource) and to choose available networks to map them on, if necessary.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:50]:
```suggestion
Liqo checks if the PodCIDR and ExternalCIDR address spaces of a remote cluster (which are listed in the `clusterconfigs.config.liqo.io` resource) are already allocated in the local cluster.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 17]
```diff
@@ -1,21 +1,98 @@
...
+The solution is implemented by the IPAM module, that keeps track of every network used in the local cluster and remaps remote clusters networks that create conflicts (traffic toward these networks will require NAT operation carried out by the [Tunnel Operator](../gateway#tunnel-operator)).
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:51]:
```suggestion
If a conflict is detected, the IPAM module, which keeps track of all networks used in the local cluster, remaps the (conflicting) networks of the remote clusters.
As a consequence, the traffic toward these networks will require an additional NAT operation in the data plane, which is carried out by the [Tunnel Operator](../gateway#tunnel-operator).
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 18]
```diff
@@ -1,21 +1,98 @@
...
+
+#### IP addresses translation of offloaded Pods
+Liqo enables to offloading of workloads on peered clusters, giving at the same time the administrator the impression Pods are running on its own cluster.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:54]:
```suggestion
Liqo enables to offloading of workloads on (remote) peered clusters, giving at the same time the illusion that all Pods are running on the local cluster.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 19]
```diff
@@ -1,21 +1,98 @@
...
+The [Virtual Kubelet](../../../offloading#virtual-kubelet) is the component in charge of offloading workloads and keep their status always up to date on the original cluster so that the administrator can inspect it.
+However, offloaded Pods are assigned an IP address by the CNI of the foreign cluster, as they actually are run there.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:56]: I would delete this line. Already said before.

File: docs/pages/concepts/networking/components/network-manager.md [hunk 20]
```diff
@@ -1,21 +1,98 @@
...
+Even if these addresses are valid on the foreign cluster, they may have no meaning in the original cluster.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:57]: I would delete this line. Already said before.

File: docs/pages/concepts/networking/components/network-manager.md [hunk 21]
```diff
@@ -1,21 +1,98 @@
...
+
+Assume that clusters A and B have both PodCIDR equal to 10.0.0.0/24 and that their administrators agree to share resources by establishing a Liqo peering.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:59]: To introduce this example, a picture with proper addresses would be extremely helpful.

File: docs/pages/concepts/networking/components/network-manager.md [hunk 22]
```diff
@@ -1,21 +1,98 @@
...
+The ExternalCIDR is the network used by Liqo for assigning addresses to external resources during Reflection(See more [Here](#reflection-of-external-endpoints)).
+{{% /notice %}}
+
+The Liqo solution is to check if networks (PodCIDR and ExternalCIDR) of a foreign cluster are already in use or are reserved in the home cluster (these networks can be specified in the `clusterconfigs.config.liqo.io` resource) and to choose available networks to map them on, if necessary.
+The solution is implemented by the IPAM module, that keeps track of every network used in the local cluster and remaps remote clusters networks that create conflicts (traffic toward these networks will require NAT operation carried out by the [Tunnel Operator](../gateway#tunnel-operator)).
+
+#### IP addresses translation of offloaded Pods
+Liqo enables to offloading of workloads on peered clusters, giving at the same time the administrator the impression Pods are running on its own cluster.
+The [Virtual Kubelet](../../../offloading#virtual-kubelet) is the component in charge of offloading workloads and keep their status always up to date on the original cluster so that the administrator can inspect it.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:55]:
```suggestion
The [Virtual Kubelet (VK)](../../../offloading#virtual-kubelet) is the component in charge of offloading workloads on the remote cluster, while keep their status always synchronized between the two clusters.
This allow the administrator of the local cluster to inspect the status of all offloaded pods (hence, running remotely) such as they were running locally.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 23]
```diff
@@ -1,21 +1,98 @@
...
+However, offloaded Pods are assigned an IP address by the CNI of the foreign cluster, as they actually are run there.
+Even if these addresses are valid on the foreign cluster, they may have no meaning in the original cluster.
+
+Assume that clusters A and B have both PodCIDR equal to 10.0.0.0/24 and that their administrators agree to share resources by establishing a Liqo peering.
+Thus, IPAM module of B decides to map A's PodCIDR on an available network, say 192.168.0.0/24.
+After the peering has been completed, B offloads Pod A1 on cluster A.
+A1 is assigned IP address 10.0.0.34 by the CNI of A, therefore the Virtual Kubelet relative to cluster A running in B detects the IP assignment and would like to update the Shadow Pod Status.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:62]:
```suggestion
If the CNI of cluster A assigns IP address 10.0.0.34 to Pod A1, the VK(A) running in B detects the IP assignment and needs to update the Shadow Pod Status.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 24]
```diff
@@ -1,21 +1,98 @@
...
+
+{{% notice info %}}
+The Shadow Pod is a local copy of the Pod resource that Liqo creates and keeps up to date when offloading a Pod.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:65]:
```suggestion
The Shadow Pod is a local copy of the Pod resource that is created and kept up to date by Liqo when offloading a Pod.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 25]
```diff
@@ -1,21 +1,98 @@
...
+{{% /notice %}}
+
+If the Virtual Kubelet updated the Shadow Pod without altering its IP, no response would have received when contacting A1 using the address specified in its Status because that IP address is not valid in cluster B and has to be changed in 192.168.0.34, according to the remapping happened during the peering phase.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:68]: I did not understand this sentence.

File: docs/pages/concepts/networking/components/network-manager.md [hunk 26]
```diff
@@ -1,21 +1,98 @@
...
+
+The IPAM module achieves this task by keeping track of how foreign clusters have remapped local networks and by providing offloaded Pod IP addresses to the Virtual Kubelet when it has to update the Status of Shadow Pods.
+
+
+#### Reflection
+One of the most relevant Liqo features is called [Reflection](../../../offloading/features/api-reflection#Overview), is carried out by the Virtual Kubelet and deals with the replication of local Kubernetes resources on foreign clusters.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:74]:
```suggestion
The [Reflection](../../../offloading/features/api-reflection#Overview) is one of the most relevant Liqo features, which is carried out by the VK and deals with the replication of local Kubernetes resources on foreign clusters.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 27]
```diff
@@ -1,21 +1,98 @@
...
+In particular, to maximize the advantages of a multi-cluster environment, the reflection of Services and Endpoints is crucial.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:75]:
```suggestion
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 28]
```diff
@@ -1,21 +1,98 @@
...
+
+In general, Endpoints that are going to be reflected can live either on:
+1. The local cluster, this is the case of Pods running on the home cluster.
+2. Elsewhere, such as a Pod not running on the home cluster or on the cluster the reflection is going to happen, but in a third one.
+
+In both cases, the IPAM module plays a decisive role, as it is in charge of providing IP addresses to reflected Endpoints.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:81]: not sure what "providing" means. Assign? Translate?

File: docs/pages/concepts/networking/components/network-manager.md (hunk 25 above)
Comment (davidefalcone1) [reply docs/pages/concepts/networking/components/network-manager.md:70]: I meant that without the translation of IP addresses of offloaded pods, entities in the home cluster trying to contact the pod using the address specified in its status would not receive any response.

File: docs/pages/concepts/networking/components/network-manager.md (hunk 20 above)
Comment (davidefalcone1) [reply docs/pages/concepts/networking/components/network-manager.md:59]: I would keep the line for 2 reasons:
1. The reader could have skipped the first section, as it was interested only on this part.
2. Since the problem is about translation of IP addresses, it is important when describing the problem to remark who assigns those addresses.

File: docs/pages/concepts/networking/components/network-manager.md (hunk 6 above)
Comment (davidefalcone1) [reply docs/pages/concepts/networking/components/network-manager.md:30]: I have deleted this sentence. 
Since the IPAM currently manages only addresses belonging to the ExternalCIDR, I will describe this point later on, probably in the 'Endpoint IP translation with ExternalCIDR' Section

File: docs/pages/concepts/networking/components/network-manager.md (hunk 25 above)
Comment (frisso) [reply docs/pages/concepts/networking/components/network-manager.md:80]:
```suggestion
In this particular example, while pod A1 has IP 10.0.0.34, its shadow copy (in cluster B) will have IP 192.168.0.34.
In fact, the former address is valid only in cluster A, while the second is valid only in clyster B; proper NATs, in the data plane, will take care of translating IP src/dest addresses accordingly in order to guarantee direct connectivity from cluster B to its offloaded pod A1.
The translation is driven by VK(A), with the help of IPAM, which determines the proper mapping for involved IP addresses.
```

Comment (davidefalcone1) [reply docs/pages/concepts/networking/components/network-manager.md:80]: I would move the sentence about the data plane in a Note section, in order to maintain the focus on the translation of addresses in the control plane.
```suggestion
In this particular example, while pod A1 has IP 10.0.0.34, its shadow copy (in cluster B) will have IP 192.168.0.34.
Since it keeps track of how remote networks have been remapped by the home cluster (See #networks-management) , the IPAM module is capable of making the translation of addresses.
For this reason, it provides an API consumed by the Virtual Kubelet when the latter has to update the Shadow Pod Status and therefore its IP address.

{{% notice note %}}
In the data plane the [Tunnel Operator](../gateway#tunnel-operator) will carry out proper NATs operations for translating IP addresses of packets coming from/directed to the offloaded Pod.
{{% /notice %}}
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 29]
```diff
@@ -1,21 +1,126 @@
...
+1. The network configuration of Cluster1 is saved in a local custom resource (`NCFG(1->2)`) containing the ClusterID of the remote cluster to whom is destined. Same step done in Cluster2;
+2. The `CRDReplicator` running in Cluster1 (which knows how to interact with the API server of Cluster2) replicates `NCFG(1->2)` in the remote cluster. The same step is done in Cluster2;
+3. The custom resource `NCFG(1->2)` in Cluster2 is processed and its status updated with the NAT information (if any). The same is done with `NCFG(2->1)` in Cluster1.
+4. The `CRDReplicator` running in Cluster1 reflects this changes in the local resource `NCFG(1->2)`. Same step done in Cluster2.
+5. The Network Manager in Cluster1 combining the status of `NCFG(1->2)` and the spec of `NCFG(2->1)` it has all the necessary information to create the custom resource `TEP(1-2)` of type TunnelEndpoint.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:37]:
```suggestion
5. The Network Manager in Cluster1 creates the custom resource `TunnelEndpoint` `TEP(1-2)` by using the information present in the status of `NCFG(1->2)` and in the spec of `NCFG(2->1)`.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 30]
```diff
@@ -1,21 +1,126 @@
...
 
 ![Liqo Networkconfigs exchange flow](../../../../images/liqonet/networkconfigs-exchange-flow.png)
 
-1. The network configuration of Cluster1 is saved in a local custom resource (`NCFG 1->2`) containing the ClusterID of the remote cluster to whom is  destined. Same step done in Cluster2;
-2. `NCFG 1->2` is replicated to the remote cluster by the `CRDReplicator`, running in Cluster1 who knows how to interact with the API server of Cluster2. Same step done in Cluster2;
-3. In Cluster2 the custom resource `NCFG (1->2)` is processed and its status updated with the NAT information if any. The same is done with `NCFG (2->1)` in Cluster1.
-4. The CRDReplicator running in Cluster1 reflects this changes in the local resource `NCFG(1->2)`. Same step done in Cluster2; The TunnelEndpointCreator in Cluster1 combining the status of `NCFG(1->2)` and the spec of `NCFG (2->1)` it has all the necessary information to create the custom resource `TEP (1-2)` of type `tunnelendpoints.net.liqo.io`.
+1. The network configuration of Cluster1 is saved in a local custom resource (`NCFG(1->2)`) containing the ClusterID of the remote cluster to whom is destined. Same step done in Cluster2;
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:33]:
```suggestion
1. The network configuration of Cluster1 is saved in a local custom resource `NCFG(1->2)`, which contains the ClusterID of the remote cluster that we want to peer with. Same step done in Cluster2.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 31]
```diff
@@ -1,21 +1,126 @@
...
+2. The `CRDReplicator` running in Cluster1 (which knows how to interact with the API server of Cluster2) replicates `NCFG(1->2)` in the remote cluster. The same step is done in Cluster2;
+3. The custom resource `NCFG(1->2)` in Cluster2 is processed and its status updated with the NAT information (if any). The same is done with `NCFG(2->1)` in Cluster1.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:35]:
```suggestion
3. Once replicated in Cluster2, the custom resource `NCFG(1->2)` is processed and its status updated with the NAT information (if any). The same is done with `NCFG(2->1)` in Cluster1.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 32]
```diff
@@ -1,21 +1,126 @@
...
+4. The `CRDReplicator` running in Cluster1 reflects this changes in the local resource `NCFG(1->2)`. Same step done in Cluster2.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:36]: What "this change" refers to?

File: docs/pages/concepts/networking/components/network-manager.md [hunk 33]
```diff
@@ -1,21 +1,126 @@
 ---
-title: Liqo-Network-Manager
+title:  Network Manager and IPAM
 weight: 1
 ---
 
-The Liqo Network Manager component manages the creation of the`networkconfigs.net.liqo.io` custom resources for the remote clusters. It also processes the`networkconfigs.net.liqo.io` received from the remote clusters remapping the remote networks if they overlap with any of the address spaces used in the local cluster. And when the`networkconfigs.net.liqo.io` exchanged by the two clusters are ready the Liqo Network Manager creates a new resource of type `tunnelendpoints.net.liqo.io` which models the network interconnection between the two clusters. The `tunnelendpoints.net.liqo.io` is reconciled by the [Tunnel Operator](../gateway#tunnel-operator) and the [Route Operator](../route#route-operator).
+The **Liqo Network Manager** is one of the Liqo components in charge of enabling peered clusters to communicate one to another.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:6]:
```suggestion
The **Liqo Network Manager** is one of the Liqo components in charge of enabling peered clusters to communicate one to another, handling the creation of the `networkconfigs.net.liqo.io` custom resources that will be sent to remote clusters.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 34]
```diff
@@ -1,21 +1,126 @@
...
+Firstly, it handles the creation of the `networkconfigs.net.liqo.io` custom resources that will be sent to remote clusters.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:7]:
```suggestion
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 35]
```diff
@@ -1,21 +1,126 @@
...
+Those special resources contain the network configuration of the home cluster: the PodCIDR, for example, is one of the fields of the Spec section of the resource.
+
+{{% notice info %}}
+The PodCIDR is the IP addressing space used by the CNI to assign addresses to Pods (e.g., _10.0.0.0/16_).
+{{% /notice %}}
+
+That is not all, the Network Manager also processes the NetworkConfigs received from the remote clusters and remap the remote networks if they overlap with any of the address spaces used in the local cluster (see [IPAM](#ipam) for more information).
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:14]:
```suggestion
In addition, the Network Manager also processes the NetworkConfigs received from the remote clusters and remaps the remote networks if they overlap with any of the address spaces used in the local cluster (see [IPAM](#ipam) for more information).
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 36]
```diff
@@ -1,21 +1,126 @@
...
+When the NetworkConfig exchanged by the two clusters are ready, the Liqo Network Manager creates a new resource of type `tunnelendpoints.net.liqo.io` which models the network interconnection between the two clusters.
+
+The following diagram illustrates how Liqo Network Manager creates the TunnelEndpoint:
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:17]:
```suggestion
The following diagram illustrates how Liqo Network Manager creates the TunnelEndpoint.
```

File: docs/pages/concepts/networking/components/network-manager.md (hunk 32 above)
Comment (davidefalcone1) [reply docs/pages/concepts/networking/components/network-manager.md:35]: The changes made on the Status section of resources. Those containing information on NAT.

File: docs/pages/concepts/networking/components/network-manager.md (hunk 29 above)
Comment (davidefalcone1) [reply docs/pages/concepts/networking/components/network-manager.md:36]: Both Spec and Status sections of resources are used.
```suggestion
5. The Network Manager in Cluster1 creates the custom resource `TunnelEndpoint` `TEP(1-2)` by using the information present in `NCFG(1->2)` and in `NCFG(2->1)`.
```

File: docs/pages/concepts/networking/components/network-manager.md (hunk 32 above)
Comment (frisso) [reply docs/pages/concepts/networking/components/network-manager.md:35]:
```suggestion
4. The `CRDReplicator` running in Cluster1 reflects the above changes back, in the local resource `NCFG(1->2)`. Same step done in Cluster2.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 37]
```diff
@@ -1,21 +1,125 @@
...
-3. In Cluster2 the custom resource `NCFG (1->2)` is processed and its status updated with the NAT information if any. The same is done with `NCFG (2->1)` in Cluster1.
-4. The CRDReplicator running in Cluster1 reflects this changes in the local resource `NCFG(1->2)`. Same step done in Cluster2; The TunnelEndpointCreator in Cluster1 combining the status of `NCFG(1->2)` and the spec of `NCFG (2->1)` it has all the necessary information to create the custom resource `TEP (1-2)` of type `tunnelendpoints.net.liqo.io`.
+1. The network configuration of Cluster1 is saved in a local custom resource `NCFG(1->2)`, which contains the ClusterID of the remote cluster that we want to peer with. Same step done in Cluster2.
+2. The `CRDReplicator` running in Cluster1 (which knows how to interact with the API server of Cluster2) replicates `NCFG(1->2)` in the remote cluster. The same step is done in Cluster2.
+3. Once replicated in Cluster2, the custom resource `NCFG(1->2)` is processed and its status updated with the NAT information (if any). The same is done with `NCFG(2->1)` in Cluster1.
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:44]:
```suggestion
3. Once replicated in Cluster2, the custom resource `NCFG(1->2)` (living in Cluster 2) is processed and its status updated with the NAT information (if any). The same is done with `NCFG(2->1)` in Cluster1.
```

File: docs/pages/concepts/networking/components/network-manager.md (hunk 32 above)
Comment (davidefalcone1) [reply docs/pages/concepts/networking/components/network-manager.md:47]:
```suggestion
4. The `CRDReplicator` running in Cluster1 reflects the updated `NCFG(1->2)` back in the local cluster. The same happens in Cluster2.
```

File: docs/pages/concepts/networking/components/network-manager.md [hunk 38]
```diff
@@ -1,21 +1,137 @@
...
+Assume that clusters A and B have both PodCIDR equal to 10.0.0.0/24 and that their administrators agree to share resources by establishing a Liqo peering.
+Thus, IPAM module of B decides to map A's PodCIDR on an available network, say 192.168.0.0/24.
+After the peering has been completed, B offloads Pod A1 on cluster A.
+If the CNI of cluster A assigns IP address 10.0.0.34 to Pod A1, the VK(A) running in B detects the IP assignment and needs to update the Shadow Pod Status.
+
+{{% notice info %}}
+The Shadow Pod is a local copy of the Pod resource that is created and kept up to date by Liqo when offloading a Pod.
+{{% /notice %}}
+
+![Example IP addresses translation when offloading a Pod](../../../../images/liqonet/offloaded-pod-ip.png?width=70%)
```
Comment (frisso) [inline docs/pages/concepts/networking/components/network-manager.md:100]:
```suggestion
![Example IP addresses translation when offloading a Pod](../../../../images/liqonet/offloaded-pod-ip.png)
```

**Review by frisso — APPROVED** [review]

**Review by palexster — APPROVED** [review]
